### PR TITLE
Feature/badge api

### PR DESCRIPTION
**Pull request by kamicut**
* Creates a run script to calculate badges for all users
* Contains `numObservations` sample metric: get all users that have had 'threshold' observations and the time at which they achieved the badge
* Fixes bug in `fetchObservations`
* Fixes seed to be a little more random

To test (after migrating)
1. Make sure to have observations in the database with new dates (run `seed-dev` again)
2. Add a badge in the badges table with the following properties:
```
id: 1
name: Pioneer
description: {"metric": "numObservations", "threshold": 10 }
```
3. Run the `calculate-badges` npm script
4. Check that `badges_users` has new entries

### CODE REVIEW

File: app/models/observations.js [hunk 1]
```diff
@@ -50,7 +50,23 @@ export async function fetchObservations (surveyId, osmObjectId, userId) {
   };
 
   const observations = await db('observations')
-    .select()
+    .select(
+      [
+        'observations.id',
+        'observations.createdAt',
+        'observations.uploadedAt',
+        'userId',
```
Comment (geohacker) [inline app/models/observations.js:58]: I suppose we should namespace these and adjust where ever these attributes are used on the frontend?

**Review by geohacker — APPROVED** [review]
This looks good to me @kamicut

File: app/models/observations.js (hunk 1 above)
Comment (kamicut) [reply app/models/observations.js:58]: Not sure how to do that with `knex` @geohacker, do you have an example?

**Comment by LanesGood** [comment]
@kamicut ready to merge?

**Comment by kamicut** [comment]
@LanesGood I've added a profile badges route so I'd like another review from @geohacker

**Review by vgeorge — APPROVED** [review]
@kamicut I changed the migration timestamp, it was set between the two campaign migrations. This is ready to merge.

**Pull request merged**